f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml

```diff
@@ -9,138 +9,456 @@
+                        "monthlyPrice": instance.monthly_price,
+                        "monthsInPeriod": monthsInPeriod(period.periodUnit, period.periodUnitCount),
+
+                    });
+                });
+            }
+            // If instance is needed for the entire task_order_duration, add it to all period_line_items
+            else if (instance.need_for_entire_task_order_duration) {
+                periods_estimate.forEach(function (period) {
+                    period.periodLineItems.push({
+                        "clin": "x001",
+                        "idiqClin": getClinFromDowTask(instance.dow_task_number, formatted_idiq_clins),
+                        "dowTaskNumber": instance.dow_task_number,
+                        "serviceOffering": service_offering.name,
+                        "itemDescriptionOrConfigSummary": instance.usage_description,
+                        "monthlyPrice": instance.monthly_price,
+                        "monthsInPeriod": monthsInPeriod(period.period.periodUnit, period.period.periodUnitCount),
+                    });
+                });
+            }
+        });
+    },
+    getClinInfo: function (idiqClins) {
+        var idiqFields = ["contract", "idiq_clin", "description", "contract_type", "classification_level"];
+        //put everything into an array
+        var clin_info = getListRecords(idiqClins.toString(), idiqFields, 'x_g_dis_atat_idiq_clin').map(removeSysId);
+        clin_info.forEach(function (item) {
+            if (item.classification_level != null) {
+                var gr = new GlideRecord('x_g_dis_atat_classification_level');
+                gr.get(item.classification_level.toString());
+                var classLevel = gr.getDisplayValue('classification');
+                item.classification_level = classLevel.toUpperCase();
+                // IGCE appends a CLASSFIED to the end of SECRET and TOP SECRET, format string to reflect that difference
+                if (item.classification_level == "SECRET" || item.classification_level == "TOP SECRET") {
+                    item.classification_level += " CLASSIFIED";
+                }
+
+            }
+        });
+
+        var formatted_clin_info = [];
+        clin_info.forEach(function (item1) {
+            clin_info.forEach(function (item2) {
+                var idiqClinString;
+                // If the items descriptions and classification_levels match, create a new idiqClinString
+                if (item1.description == item2.description && item1.classification_level == item2.classification_level && item1.idiq_clin != item2.idiq_clin) {
+                    // Lowest IDIQ CLIN first
+                    if (parseInt(item1.idiq_clin) < parseInt(item2.idiq_clin)) {
+                        idiqClinString = "x" + item1.idiq_clin.substr(1) + "/x" + item2.idiq_clin.substr(1) + " " + item1.description + " " + item1.classification_level;
+                    }
```

Comment:
Is two `forEach` loops needed on `clin_info`? I don't completely understand what is happening by having two. Could sorting prior to the `forEach` help?

Reply:
@zachclark-ccpo Can you provide some insight here?  It looks like you are comparing two CLINs to determine whether an IDIQ CLIN is needed.